src/main/setup.ts

```diff
@@ -186,4 +190,31 @@ export default function setupMainIpc({ dolphinManager }: { dolphinManager: Dolph
     const { canceled, filePaths } = await dialog.showOpenDialog(options);
     return { canceled, filePaths };
   });
+
+  ipc_runDiagnosticNat.main!.handle(async () => {
+    try {
+      const result = await getNatType();
+      return result;
+    } catch {
+      return { address: "", natType: NatType.FAILED };
+    }
+  });
+
+  ipc_runDiagnosticPortMapping.main!.handle(async () => {
+    try {
+      const result = await getPortMappingPresence();
+      return result;
+    } catch {
+      return { upnp: Presence.FAILED, natpmp: Presence.FAILED };
+    }
+  });
+
+  ipc_runDiagnosticCgnat.main!.handle(async ({ address }) => {
+    try {
+      const result = await getCgnatPresence(address);
+      return result;
+    } catch {
+      return { cgnat: Presence.FAILED };
+    }
+  });
```

Comment:
We don't need 3 different IPC endpoints when they're always called together. Just have a single IPC endpoint that fetches and returns all the necessary info required to render the FE.